src/mesh/boundary_info.C

```diff
@@ -1438,9 +1490,6 @@ void BoundaryInfo::remove_side (const Elem * elem,
 {
   libmesh_assert(elem);
 
-  // Only level 0 elements are stored in BoundaryInfo.
-  libmesh_assert_equal_to (elem->level(), 0);
-
   // Erase (elem, side, *) entries from map.
   erase_if(_boundary_side_id, elem,
            [side](decltype(_boundary_side_id)::mapped_type & pr)
```

Comment:
Okay, I think my only remaining concern is what came to me on Slack yesterday:

> We actually probably ought to throw an error for that remove_side() case, if we're not going to support it yet, the more I think about it.  If a child has id 5 because its parent does, and we try to remove it from the child, then ideally we remove it from the parent and restore it on the other children, less ideally we exit with an error and make the app handle it, but what we do now is silently do nothing, and that's totally wrong.

If someone tries to remove a side that isn't there, our current do-nothing behavior is fine, it's like putting white paint on an already-white wall.  But if someone tries to remove a side that is there via an ancestor, but *isn't a raw side*, our current do-nothing behavior isn't fine.  Put white paint on a blue wall and you don't expect to see blue afterwards, not even if the blue was put on with a large brush and the white is being put on with a smaller brush.

This is a new feature, so it doesn't need to be a *comprehensive* new feature, but we definitely don't want to set up applications for failure if the author mistakenly thinks it's comprehensive or if the author accidentally starts to assume that the missing parts of the feature are missing by design.

Could we modify erase_if to return a std::size_t counting the number of erasures, then (here and in the bcid-specific case below) if we get a 0 from that we'll double-check that `boundary_ids(elem, side, vec)` is empty (or below, that `has_boundary_id(elem, side, id)` is false), then throw `libmesh_not_implemented()` if so?